js/js.libraries/src/core/collections.kt

```diff
@@ -112,7 +133,7 @@ private fun <T> collectionsSort(list: MutableList<T>, comparator: Comparator<in
 
     array.asDynamic().sort(comparator.asDynamic().compare.bind(comparator))
 
-    for (i in 0..array.size - 1) {
+    for (i in array.indices) {
```

Comment:
JS backend doesn't optimize such loops ATM.

Reply:
Thanks, changed it to `0..array.lastIndex`